Document the requirement for a transaction for stateful repositories in Javadoc (#1429)
jakartaee/data: spec/src/main/asciidoc/repository.asciidoc

- Multiple repositories might share a persistence context, especially if they share a datastore.

> - A persistence context is never shared across transactions that are active at the same time.

Reviewer: ```suggestion
- A persistence context is never shared across concurrent transactions.
```
